### ISSUE TITLE
WebUI: after reinstalling WebUI the Apps page is not openning

### ISSUE THREAD

**Issue by 1thales** [raspiblitz/raspiblitz#3331]
So i deleted webUI then decided to reenable it again. Homepage is opening but when i go to the apps page i am met with a forever rotating wheel :(

**Comment by ChuckNorrison** [comment]
Happened to me after reinstall from blitz menu, no solution found yet

**Comment by cstenglein** [comment]
If you update the api & webUI, it should be fixed now:

```
sudo ./config.scripts/blitz.web.api.sh update-code
sudo ./config.scripts/blitz.web.ui.sh update
```

Otherwise fixed with the next version.

Problem was the branches were broken, we will change the branching strategy with the next version, so it doesn't break midway.

**Comment by 1thales** [comment]
this is what i get when i run the first command

admin@192.168.0.93:~ ₿ sudo ./config.scripts/blitz.web.api.sh update-code
ls: cannot access '/etc/systemd/system/blitzapi.service': No such file or directory
#blitzapi not active

**Comment by cstenglein** [comment]
> this is what i get when i run the first command

looks like you uninstalled the API, as seen by the comment:

>  #blitzapi not active

Then you can use:

```sudo ./config.scripts/blitz.web.api.sh on```

**Comment by 1thales** [comment]
`admin@192.168.0.93:~ ₿ sudo ./config.scripts/blitz.web.api.sh on
# INSTALL Web API ...
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LC_CTYPE = "UTF-8",
	LANG = "en_GB.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("en_GB.UTF-8").
adduser: The user `blitzapi' already exists.
# allowing user as part of the bitcoin group to RW RPC hook
chmod: cannot access '/home/bitcoin/.lightning/bitcoin': No such file or directory
chmod: cannot access '/home/bitcoin/.lightning/bitcoin/lightning-rpc': No such file or directory
cat: /home/bitcoin/.lightning/config: No such file or directory
tee: /home/bitcoin/.lightning/config: No such file or directory
rpc-file-mode=0660
Cloning into '/home/blitzapi/blitz_api'...
remote: Enumerating objects: 2199, done.
remote: Counting objects: 100% (794/794), done.
remote: Compressing objects: 100% (307/307), done.
remote: Total 2199 (delta 602), reused 617 (delta 485), pack-reused 1405
Receiving objects: 100% (2199/2199), 1.23 MiB | 3.99 MiB/s, done.
Resolving deltas: 100% (1487/1487), done.
Already on 'main'
Your branch is up to date with 'origin/main'.
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Ignoring cffi: markers 'python_version >= "3.9" and python_version < "4.0" and implementation_name == "pypy"' don't match your environment
Ignoring colorama: markers 'python_version >= "3.9" and python_version < "4.0" and platform_system == "Windows"' don't match your environment
Ignoring py: markers 'python_version >= "3.9" and python_version < "4.0" and implementation_name == "pypy"' don't match your environment
Ignoring pycparser: markers 'python_version >= "3.9" and python_version < "4.0" and implementation_name == "pypy"' don't match your environment
Requirement already satisfied: aiohttp==3.8.1 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 1)) (3.8.1)
Requirement already satisfied: aiojobs==1.0.0 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 74)) (1.0.0)
Collecting aioredis==2.0.1
  Downloading https://www.piwheels.org/simple/aioredis/aioredis-2.0.1-py3-none-any.whl (71 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 71.2/71.2 kB 776.9 kB/s eta 0:00:00
Requirement already satisfied: aiosignal==1.2.0 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 85)) (1.2.0)
Requirement already satisfied: anyio==3.6.1 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 89)) (3.6.1)
Requirement already satisfied: async-timeout==4.0.2 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 93)) (4.0.2)
Collecting attrs==22.1.0
  Downloading https://www.piwheels.org/simple/attrs/attrs-22.1.0-py2.py3-none-any.whl (58 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.8/58.8 kB 1.1 MB/s eta 0:00:00
Requirement already satisfied: cchardet==2.1.7 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 101)) (2.1.7)
Collecting certifi==2022.9.24
  Downloading https://www.piwheels.org/simple/certifi/certifi-2022.9.24-py3-none-any.whl (161 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 161.1/161.1 kB 3.6 MB/s eta 0:00:00
Collecting charset-normalizer==2.1.1
  Downloading https://www.piwheels.org/simple/charset-normalizer/charset_normalizer-2.1.1-py3-none-any.whl (39 kB)
Requirement already satisfied: click==8.1.3 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 204)) (8.1.3)
Requirement already satisfied: deepdiff==5.8.1 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 211)) (5.8.1)
Collecting deprecated==1.2.13
  Downloading https://www.piwheels.org/simple/deprecated/Deprecated-1.2.13-py2.py3-none-any.whl (9.5 kB)
Collecting fastapi-plugins==0.10.0
  Downloading https://www.piwheels.org/simple/fastapi-plugins/fastapi_plugins-0.10.0-py3-none-any.whl (23 kB)
Collecting fastapi==0.82.0
  Using cached https://www.piwheels.org/simple/fastapi/fastapi-0.82.0-py3-none-any.whl (55 kB)
Collecting frozenlist==1.3.1
  Using cached frozenlist-1.3.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (159 kB)
Collecting googleapis-common-protos==1.56.4
  Downloading https://www.piwheels.org/simple/googleapis-common-protos/googleapis_common_protos-1.56.4-py2.py3-none-any.whl (211 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 211.7/211.7 kB 1.9 MB/s eta 0:00:00
Collecting grpcio-tools==1.48.1
  Using cached grpcio_tools-1.48.1-cp39-cp39-manylinux_2_17_aarch64.whl (31.5 MB)
Collecting grpcio==1.48.1
  Using cached grpcio-1.48.1-cp39-cp39-manylinux_2_17_aarch64.whl (63.9 MB)
Collecting h11==0.14.0
  Downloading https://www.piwheels.org/simple/h11/h11-0.14.0-py3-none-any.whl (58 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.3/58.3 kB 732.2 kB/s eta 0:00:00
Requirement already satisfied: hiredis==2.0.0 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 388)) (2.0.0)
Collecting idna==3.4
  Downloading https://www.piwheels.org/simple/idna/idna-3.4-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.5/61.5 kB 1.4 MB/s eta 0:00:00
Requirement already satisfied: multidict==6.0.2 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 434)) (6.0.2)
Requirement already satisfied: ordered-set==4.1.0 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 494)) (4.1.0)
Requirement already satisfied: packaging==21.3 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 498)) (21.3)
Collecting protobuf==3.20.3
  Using cached protobuf-3.20.3-cp39-cp39-manylinux2014_aarch64.whl (918 kB)
Collecting psutil==5.9.2
  Using cached psutil-5.9.2.tar.gz (479 kB)
  Preparing metadata (setup.py) ... done
Collecting pydantic==1.10.2
  Using cached pydantic-1.10.2-py3-none-any.whl (154 kB)
Requirement already satisfied: pyjwt==2.4.0 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 601)) (2.4.0)
Requirement already satisfied: pyparsing==3.0.9 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 605)) (3.0.9)
Requirement already satisfied: python-decouple==3.6 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 608)) (3.6)
Collecting python-json-logger==2.0.4
  Downloading https://www.piwheels.org/simple/python-json-logger/python_json_logger-2.0.4-py3-none-any.whl (7.8 kB)
Collecting pyzmq==23.2.1
  Using cached pyzmq-23.2.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (888 kB)
Collecting redis==4.3.4
  Using cached redis-4.3.4-py3-none-any.whl (246 kB)
Collecting requests==2.28.1
  Downloading https://www.piwheels.org/simple/requests/requests-2.28.1-py3-none-any.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.8/62.8 kB 932.9 kB/s eta 0:00:00
Collecting setuptools==65.4.1
  Downloading https://www.piwheels.org/simple/setuptools/setuptools-65.4.1-py3-none-any.whl (1.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 2.3 MB/s eta 0:00:00
Requirement already satisfied: six==1.16.0 in /usr/lib/python3/dist-packages (from -r requirements.txt (line 702)) (1.16.0)
Collecting sniffio==1.3.0
  Downloading https://www.piwheels.org/simple/sniffio/sniffio-1.3.0-py3-none-any.whl (10 kB)
Requirement already satisfied: starlette==0.19.1 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 709)) (0.19.1)
Collecting tenacity==8.1.0
  Downloading https://www.piwheels.org/simple/tenacity/tenacity-8.1.0-py3-none-any.whl (23 kB)
Collecting typing-extensions==4.3.0
  Downloading https://www.piwheels.org/simple/typing-extensions/typing_extensions-4.3.0-py3-none-any.whl (25 kB)
Collecting urllib3==1.26.12
  Downloading https://www.piwheels.org/simple/urllib3/urllib3-1.26.12-py2.py3-none-any.whl (140 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 140.4/140.4 kB 4.1 MB/s eta 0:00:00
Collecting uvicorn==0.18.3
  Using cached https://www.piwheels.org/simple/uvicorn/uvicorn-0.18.3-py3-none-any.whl (57 kB)
Collecting wrapt==1.14.1
  Using cached wrapt-1.14.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (77 kB)
Requirement already satisfied: yarl==1.8.1 in /usr/local/lib/python3.9/dist-packages (from -r requirements.txt (line 793)) (1.8.1)
Building wheels for collected packages: psutil
  Building wheel for psutil (setup.py) ... done
  Created wheel for psutil: filename=psutil-5.9.2-cp39-cp39-linux_aarch64.whl size=283984 sha256=8ad573fd755a8cd45edd05b08b71cf0b1dea899f40e6ec29ac1ba161c596bb58
  Stored in directory: /root/.cache/pip/wheels/90/b2/61/285f7409a9dd2ffa1aa0684602a619f61b4c7b41c7ba7ba7e3
Successfully built psutil
Installing collected packages: wrapt, uvicorn, urllib3, typing-extensions, tenacity, sniffio, setuptools, requests, redis, pyzmq, python-json-logger, pydantic, psutil, protobuf, idna, h11, grpcio-tools, grpcio, googleapis-common-protos, frozenlist, fastapi-plugins, fastapi, deprecated, charset-normalizer, certifi, attrs, aioredis
  Attempting uninstall: wrapt
    Found existing installation: wrapt 1.12.1
ERROR: Cannot uninstall 'wrapt'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.
--- Logging error ---
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/pip/_internal/utils/logging.py", line 177, in emit
    self.console.print(renderable, overflow="ignore", crop=False, style=style)
  File "/usr/local/lib/python3.9/dist-packages/pip/_vendor/rich/console.py", line 1752, in print
    extend(render(renderable, render_options))
  File "/usr/local/lib/python3.9/dist-packages/pip/_vendor/rich/console.py", line 1390, in render
    for render_output in iter_render:
  File "/usr/local/lib/python3.9/dist-packages/pip/_internal/utils/logging.py", line 134, in __rich_console__
    for line in lines:
  File "/usr/local/lib/python3.9/dist-packages/pip/_vendor/rich/segment.py", line 245, in split_lines
    for segment in segments:
  File "/usr/local/lib/python3.9/dist-packages/pip/_vendor/rich/console.py", line 1368, in render
    renderable = rich_cast(renderable)
  File "/usr/local/lib/python3.9/dist-packages/pip/_vendor/rich/protocol.py", line 36, in rich_cast
    renderable = cast_method()
  File "/usr/local/lib/python3.9/dist-packages/pip/_internal/self_outdated_check.py", line 130, in __rich__
    pip_cmd = get_best_invocation_for_this_pip()
  File "/usr/local/lib/python3.9/dist-packages/pip/_internal/utils/entrypoints.py", line 58, in get_best_invocation_for_this_pip
    if found_executable and os.path.samefile(
  File "/usr/lib/python3.9/genericpath.py", line 101, in samefile
    s2 = os.stat(f2)
FileNotFoundError: [Errno 2] No such file or directory: '/usr/bin/pip3.9'
Call stack:
  File "/usr/local/bin/pip", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/dist-packages/pip/_internal/cli/main.py", line 70, in main
    return command.main(cmd_args)
  File "/usr/local/lib/python3.9/dist-packages/pip/_internal/cli/base_command.py", line 101, in main
    return self._main(args)
  File "/usr/local/lib/python3.9/dist-packages/pip/_internal/cli/base_command.py", line 223, in _main
    self.handle_pip_version_check(options)
  File "/usr/local/lib/python3.9/dist-packages/pip/_internal/cli/req_command.py", line 148, in handle_pip_version_check
    pip_self_version_check(session, options)
  File "/usr/local/lib/python3.9/dist-packages/pip/_internal/self_outdated_check.py", line 237, in pip_self_version_check
    logger.info("[present-rich] %s", upgrade_prompt)
  File "/usr/lib/python3.9/logging/__init__.py", line 1442, in info
    self._log(INFO, msg, args, **kwargs)
  File "/usr/lib/python3.9/logging/__init__.py", line 1585, in _log
    self.handle(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 1595, in handle
    self.callHandlers(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 1657, in callHandlers
    hdlr.handle(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 948, in handle
    self.emit(record)
  File "/usr/local/lib/python3.9/dist-packages/pip/_internal/utils/logging.py", line 179, in emit
    self.handleError(record)
Message: '[present-rich] %s'
Arguments: (UpgradePrompt(old='22.1.2', new='22.3.1'),)
error='pip install failed'`

**Comment by 1thales** [comment]
didn't work :(

**Comment by ChuckNorrison** [comment]
faced the same errors here
https://github.com/rootzoll/raspiblitz/issues/3262#issuecomment-1272602987

Its still unknown how this happened. Fix manually as described or reflash sd card works for me.

**Comment by cstenglein** [comment]
@rootzoll @fusion44 can we fix the pip issue in the scripts / the API?

Also occurred on my test node.

**Comment by rootzoll** [comment]
@cstenglein does this problem still exists when testing with the new dev image?

**Comment by cstenglein** [comment]
No, issue was fixed some time ago in the blitzAPI.

Main issue for using that fix was https://github.com/rootzoll/raspiblitz/pull/3466 .

Can be closed now or with 1.9.0 release
